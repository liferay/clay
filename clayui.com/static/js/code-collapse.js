function closestAncestor(node, s) {
	const el = node;
	let ancestor = node;

	if (!document.documentElement.contains(el)) {
		return null;
	}

	do {
		if (ancestor.matches(s)) {
			return ancestor;
		}

		ancestor = ancestor.parentElement;
	} while (ancestor !== null);

	return null;
}

if (document.body) {
	document.body.addEventListener('click', event => {
		const target = event.target;
		const targetClassList = target.classList;

		if (
			targetClassList.contains('btn-collapse') ||
			targetClassList.contains('icon-collapse') ||
			(target.tagName === 'use' &&
				target.parentNode.classList.contains('icon-collapse'))
		) {
			const codeContainer = closestAncestor(target, '.code-container');

			if (codeContainer) {
				const gatsbyCode = codeContainer.querySelector(
					'code[class^="gatsby-code-"]'
				);

				if (gatsbyCode) {
					const btnCollapse = codeContainer.querySelector(
						'.btn-collapse--collapse'
					);
					const btnExpand = codeContainer.querySelector(
						'.btn-collapse--expand'
					);

					if (gatsbyCode.style.display === 'none') {
						btnCollapse.classList.remove('hide');
						btnExpand.classList.add('hide');

						gatsbyCode.style.display = 'block';
					} else {
						btnCollapse.classList.add('hide');
						btnExpand.classList.remove('hide');

						gatsbyCode.style.display = 'none';
					}
				}
			}
		}
	});
}
