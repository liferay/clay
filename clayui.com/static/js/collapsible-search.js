/**
* Clay 2.10.0
*
* Copyright 2019, Liferay, Inc.
* All rights reserved.
* MIT license
*/
+function($) {
	var CollapsibleSearch = function(element) {
		var instance = this;

		instance.$element = $(element);
		instance.$close = instance.$element.find('.basic-search-close');
		instance.$input = instance.$element.find('input[type="text"]');
		instance.$submit = instance.$element.find('[type="submit"]');

		instance.$close.on('click.lexicon.close.collapsible-search', $.proxy(instance.close, instance));
		instance.$input.on('blur.lexicon.collapsible-search', $.proxy(instance.blur, instance));
		instance.$input.on('focus.lexicon.collapsible-search', $.proxy(instance.focus, instance));
		instance.$submit.on('click.lexicon.submit.collapsible-search', $.proxy(instance.submit, instance));
	};

	CollapsibleSearch.BREAKPOINT = 768;
	CollapsibleSearch.TRANSITION_DURATION = 500;

	CollapsibleSearch.prototype = {
		blur: function(event) {
			var $input = $(event.currentTarget);

			$input.closest('.basic-search').removeClass('focus');
		},

		close: function(event) {
			var instance = this;

			var basicSearch = $(event.currentTarget).closest('.basic-search');

			var basicSearchSlider = basicSearch.find('.basic-search-slider');
			var basicSearchSubmit = basicSearch.find('[type="submit"]');

			var complete = function() {
				basicSearch.removeClass('basic-search-transition');

				basicSearch.trigger('closed.lexicon.collapsible.search');
			};

			var supportsTransition = bootstrap.Util.supportsTransitionEnd();

			if (supportsTransition) {
				basicSearchSlider.one('bsTransitionEnd', $.proxy(complete, instance))
					.emulateTransitionEnd(CollapsibleSearch.TRANSITION_DURATION);
			}

			basicSearch.addClass('basic-search-transition').removeClass('open');

			if (!supportsTransition) {
				complete.call(instance);
			}
			else {
				basicSearchSubmit.focus();
			}
		},

		destroy: function() {
			var instance = this;

			instance.$close.off('click.lexicon.close.collapsible-search');
			instance.$input.off('blur.lexicon.collapsible-search');
			instance.$input.off('focus.lexicon.collapsible-search');
			instance.$submit.off('click.lexicon.submit.collapsible-search');
		},

		focus: function(event) {
			$(event.currentTarget).closest('.basic-search').addClass('focus');
		},

		submit: function(event) {
			var instance = this;

			if (window.innerWidth < CollapsibleSearch.BREAKPOINT) {
				var basicSearch = $(event.currentTarget).parents('.basic-search');

				var basicSearchInput = basicSearch.find('input[type="text"]');
				var basicSearchSlider = basicSearch.find('.basic-search-slider');

				var complete = function() {
					basicSearch.removeClass('basic-search-transition');
					basicSearchInput.focus();

					basicSearch.trigger('open.lexicon.collapsible.search');
				};

				if (!basicSearch.hasClass('open')) {
					event.preventDefault();

					var supportsTransition = bootstrap.Util.supportsTransitionEnd();

					if (supportsTransition) {
						basicSearchSlider.one('bsTransitionEnd', $.proxy(complete, instance))
							.emulateTransitionEnd(CollapsibleSearch.TRANSITION_DURATION);
					}

					basicSearch.addClass('basic-search-transition').addClass('open');

					if (!supportsTransition) {
						complete.call(instance);
					}
				}
			}
		}
	};

	var Plugin = function(option) {
		return this.each(
			function() {
				var $this = $(this);

				var data = $this.data('lexicon.collapsible-search');

				if (!data) {
					data = new CollapsibleSearch(this);

					$this.data('lexicon.collapsible-search', data);
				}

				if (typeof option == 'string') {
					data[option]();
				}
			}
		);
	};

	var old = $.fn.collapsibleSearch;

	$.fn.collapsibleSearch = Plugin;
	$.fn.collapsibleSearch.Constructor = CollapsibleSearch;

	$.fn.collapsibleSearch.noConflict = function() {
		$.fn.collapsibleSearch = old;

		return this;
	};

	var close = '[data-toggle="collapsible-search"] .basic-search-close';
	var input = '[data-toggle="collapsible-search"] input[type="text"]';
	var submit = '[data-toggle="collapsible-search"] [type="submit"]';

	$(document)
		.on('blur.lexicon.collapsible-search.data-api', input, $.proxy(CollapsibleSearch.prototype.blur, CollapsibleSearch))
		.on('click.lexicon.close.collapsible-search.data-api', close, $.proxy(CollapsibleSearch.prototype.close, CollapsibleSearch))
		.on('click.lexicon.submit.collapsible-search.data-api', submit, $.proxy(CollapsibleSearch.prototype.submit, CollapsibleSearch))
		.on('focus.lexicon.collapsible-search.data-api', input, $.proxy(CollapsibleSearch.prototype.focus, CollapsibleSearch));
}(jQuery);