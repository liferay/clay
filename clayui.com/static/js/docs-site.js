if (!Element.prototype.matches) {
	Element.prototype.matches = 
		Element.prototype.msMatchesSelector ||
		Element.prototype.webkitMatchesSelector;
}

if (!Element.prototype.closest) {
	Element.prototype.closest = function(selector) {
		var node = this;

		while (node.nodeType === 1) {
			if (node.matches(selector)) {
				return node;
			}

			node = node.parentNode;
		}

		return null;
	}
}

(function () {
	function toggleCodeCollapse(e) {
		var isCodeCollapse =
			e.target.classList.contains('btn-collapse--collapse') ||
			e.target.classList.contains('btn-collapse--expand') ||
			e.target.closest('.btn-collapse--collapse') ||
			e.target.closest('.btn-collapse--expand');

		if (isCodeCollapse) {
			var codeContainer = e.target.closest('.code-container');
			var gatsbyCode = codeContainer.querySelector(
				'.gatsby-highlight > pre'
			);

			var isVisible =
				window.getComputedStyle(gatsbyCode).display !== 'none';

			var display = isVisible ? 'none' : 'block';
			var focusClass = isVisible
				? '.btn-collapse--expand'
				: '.btn-collapse--collapse';

			codeContainer
				.querySelector('.btn-collapse--collapse')
				.classList.toggle('hide');
			codeContainer
				.querySelector('.btn-collapse--expand')
				.classList.toggle('hide');
			codeContainer.querySelector(focusClass).focus();
			gatsbyCode.style.display = display;
		}
	}

	document.addEventListener('click', toggleCodeCollapse);
})();

(function() {
	document.addEventListener('click', function(event) {
		var t = event.target;

		var a = t.tagName === 'a' ? t : t.closest('a');

		if (a) {
			if (a.getAttribute('href') === '#1') {
				event.preventDefault();
			}
		}
	});
})();
