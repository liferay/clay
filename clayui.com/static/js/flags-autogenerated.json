[
    {
        "name": "ar-sa",
        "aliases": ["ar-sa", "Arabic", "Saudi Arabia"]
    },
    {
        "name": "bg-bg",
        "aliases": ["bg-bg", "Bulgarian", "Bulgaria"]
    },
    {
        "name": "ca-ad",
        "aliases": ["ca-ad", "Catalan", "Andorra"]
    },
    {
        "name": "ca-es",
        "aliases": ["ca-es", "Catalan", "Spain"]
    },
    {
        "name": "cs-cz",
        "aliases": ["cs-cz", "Czech", "Czech Republic"]
    },
    {
        "name": "da-dk",
        "aliases": ["da-dk", "Danish", "Denmark"]
    },
    {
        "name": "de-de",
        "aliases": ["de-de", "German", "Germany"]
    },
    {
        "name": "el-gr",
        "aliases": ["el-gr", "Greek", "Greece"]
    },
    {
        "name": "en-au",
        "aliases": ["en-au", "English", "Australia"]
    },
    {
        "name": "en-gb",
        "aliases": ["en-gb", "English", "United Kingdom"]
    },
    {
        "name": "en-us",
        "aliases": ["en-us", "English", "United States"]
    },
    {
        "name": "es-es",
        "aliases": ["es-es", "Spanish", "Spain"]
    },
    {
        "name": "et-ee",
        "aliases": ["et-ee", "Estonian", "Estonia"]
    },
    {
        "name": "eu-es",
        "aliases": ["eu-es", "Basque", "Spain"]
    },
    {
        "name": "fa-ir",
        "aliases": ["fa-ir", "Farsi", "Iran"]
    },
    {
        "name": "fi-fi",
        "aliases": ["fi-fi", "Finnish", "Finland"]
    },
    {
        "name": "fr-ca",
        "aliases": ["fr-ca", "French", "Canada"]
    },
    {
        "name": "fr-fr",
        "aliases": ["fr-fr", "French", "France"]
    },
    {
        "name": "gl-es",
        "aliases": ["gl-es", "Galician", "Spain"]
    },
    {
        "name": "hi-in",
        "aliases": ["hi-in", "Hindi", "India"]
    },
    {
        "name": "hr-hr",
        "aliases": ["hr-hr", "Croatian", "Croatia"]
    },
    {
        "name": "hu-hu",
        "aliases": ["hu-hu", "Hungarian", "Hungary"]
    },
    {
        "name": "in-id",
        "aliases": ["in-id", "Indonesian", "Indonesia"]
    },
    {
        "name": "it-it",
        "aliases": ["it-it", "Italian", "Italy"]
    },
    {
        "name": "iw-il",
        "aliases": ["iw-il", "Hebrew", "Israel"]
    },
    {
        "name": "ja-jp",
        "aliases": ["ja-jp", "Japanese", "Japan"]
    },
    {
        "name": "kk-kz",
        "aliases": ["kk-kz", "Kazakh", "Kazakhstan"]
    },
    {
        "name": "ko-kr",
        "aliases": ["ko-kr", "Korean", "South Korea"]
    },
    {
        "name": "lo-la",
        "aliases": ["lo-la", "Lao", "Laos"]
    },
    {
        "name": "lt-lt",
        "aliases": ["lt-lt", "Lithuanian", "Lithuania"]
    },
    {
        "name": "ms-my",
        "aliases": ["ms-my", "Malay", "Malaysia"]
    },
    {
        "name": "nb-no",
        "aliases": ["nb-no", "Norwegian Bokm√•l", "Norway"]
    },
    {
        "name": "nl-be",
        "aliases": ["nl-be", "Dutch", "Belgium"]
    },
    {
        "name": "nl-nl",
        "aliases": ["nl-nl", "Dutch", "Netherlands"]
    },
    {
        "name": "pl-pl",
        "aliases": ["pl-pl", "Polish", "Poland"]
    },
    {
        "name": "pt-br",
        "aliases": ["pt-br", "Portuguese", "Brazil"]
    },
    {
        "name": "pt-pt",
        "aliases": ["pt-pt", "Portuguese", "Portugal"]
    },
    {
        "name": "ro-ro",
        "aliases": ["ro-ro", "Romanian", "Romania"]
    },
    {
        "name": "ru-ru",
        "aliases": ["ru-ru", "Russian", "Russia"]
    },
    {
        "name": "sk-sk",
        "aliases": ["sk-sk", "Slovak", "Slovakia"]
    },
    {
        "name": "sl-si",
        "aliases": ["sl-si", "Slovenian", "Slovenia"]
    },
    {
        "name": "sr-rs-latin",
        "aliases": ["sr-rs-latin", "Serbian Latin", "Serbia"]
    },
    {
        "name": "sr-rs",
        "aliases": ["sr-rs", "Serbian", "Serbia"]
    },
    {
        "name": "sv-se",
        "aliases": ["sv-se", "Swedish", "Sweden"]
    },
    {
        "name": "ta-in",
        "aliases": ["ta-in", "Tamil", "India"]
    },
    {
        "name": "th-th",
        "aliases": ["th-th", "Thai", "Thailand"]
    },
    {
        "name": "tr-tr",
        "aliases": ["tr-tr", "Turkish", "Turkey"]
    },
    {
        "name": "uk-ua",
        "aliases": ["uk-ua", "Ukrainian", "Ukraine"]
    },
    {
        "name": "vi-vn",
        "aliases": ["vi-vn", "Vietnamese", "Vietnam"]
    },
    {
        "name": "zh-cn",
        "aliases": ["zh-cn", "Chinese", "China"]
    },
    {
        "name": "zh-tw",
        "aliases": ["zh-tw", "Chinese", "Taiwan"]
    }
]