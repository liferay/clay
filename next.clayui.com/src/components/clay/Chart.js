/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import React from 'react';

import mapData from '../../../static/map.json';
import Editor from './Editor';

// Workaround for building gatsby's SSR with since Billboard.js references `window`
const ClayChart =
	typeof window !== `undefined`
		? require('@clayui/charts').default
		: () => <div />;

const chartImportsCode = `import ClayChart from '@clayui/charts';
import React from 'react';`;

const chartCode = `const Component = () => {
	const [active, setActive] = React.useState(false);

	const COLUMNS = [['data1', 100, 20, 30], ['data2', 20, 70, 100]];
	const COLUMNS_2 = [['data1', 10, 50, 60], ['data2', 70, 30, 10]];

	return (
		<div>
			<button onClick={() => setActive(val => !val)}>{'Change Data'}</button>

			<ClayChart
				data={{
					columns: active ? COLUMNS : COLUMNS_2,
					type: 'bar',
				}}
			/>
		</div>
	);
}

render(<Component />);`;

export const Chart = () => {
	const scope = {ClayChart};
	const code = chartCode;

	return <Editor code={code} imports={chartImportsCode} scope={scope} />;
};

const geoMapImportsCode = `import ClayChart from '@clayui/charts';
import React from 'react';
import mapData from './map.json';`;

const geoMapCode = `const Component = () => (
	<div style={{height: 380}}>
		<ClayChart
			color={{
				range: {
					max: '#0065e4',
					min: '#b1d4ff',
				},
				selected: '#4b9bff',
				value: 'gdp_md_est',
			}}
			data={{
				...mapData,
				type: 'geo-map',
			}}
		/>
	</div>
);

render(<Component />);`;

export const GeoMap = () => {
	const scope = {ClayChart, mapData};
	const code = geoMapCode;

	return <Editor code={code} imports={geoMapImportsCode} scope={scope} />;
};

const predictiveImportsCode = `import ClayChart from '@clayui/charts';
import React from 'react';`;

const predictiveCode = `const Component = () => (
	<ClayChart
		axis={{
			x: {
				type: 'timeseries',
			},
		}}
		data={{
			columns: [
				[
					'x',
					'2018-01-01',
					'2018-02-01',
					'2018-03-01',
					'2018-04-01',
					'2018-05-01',
					'2018-06-01',
					'2018-07-01',
					'2018-08-01',
					'2018-09-01',
					'2018-10-01',
					'2018-11-01',
					'2018-12-01',
				],
				[
					'product1',
					130,
					340,
					200,
					100,
					40,
					300,
					{high: 240, low: 140, mid: 180},
					{high: 380, low: 300, mid: 350},
					{high: 480, low: 320, mid: 400},
					{high: 260, low: 100, mid: 200},
					{high: 140, low: 100, mid: 120},
					{high: 180, low: 80, mid: 100},
				],
				[
					'product2',
					210,
					180,
					30,
					90,
					40,
					120,
					{high: 260, low: 180, mid: 240},
					{high: 460, low: 360, mid: 420},
					{high: 180, low: 80, mid: 120},
					{high: 120, low: 60, mid: 80},
					{high: 80, low: 10, mid: 20},
					{high: 100, low: 20, mid: 60},
				],
			],
			type: 'predictive',
			types: {
				product1: 'area-line-range',
				product2: 'area-spline-range',
			},
			x: 'x',
		}}
		predictionDate="2018-06-01"
	/>
);

render(<Component />);`;

export const Predictive = () => {
	const scope = {ClayChart};
	const code = predictiveCode;

	return <Editor code={code} imports={predictiveImportsCode} scope={scope} />;
};
