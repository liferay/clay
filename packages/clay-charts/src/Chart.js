import ChartBase from './ChartBase';
import Component from 'metal-component';
import {Config} from 'metal-state';
import Soy from 'metal-soy';
import templates from './Chart.soy.js';

/**
 * Metal Chart component.
 * @extends ChartBase
 * @extends Component
 */
class Chart extends Component {}

Object.assign(Chart.prototype, ChartBase);

Chart.STATE = {
	/**
	 * Id to be applied to the element.
	 * @default undefined
	 * @instance
	 * @memberof Chart
	 * @type {?(string|undefined)}
	 */
	id: Config.string(),
};

Chart.STATE = Object.assign(Chart.STATE, ChartBase.STATE);

Soy.register(Chart, templates);

export {Chart};
export default Chart;
