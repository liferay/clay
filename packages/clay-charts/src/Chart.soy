{namespace ClayChart}

/**
 * This renders the component's whole content.
 */
{template .render}
	{@param? _loading:  bool}

	<div class="metal-chart-container">
		{let $chartAttrs kind="attributes"}
			{if $_loading}
				hidden
			{/if}

			ref="chart"
		{/let}

		<div class="metal-chart" {$chartAttrs}></div>

		{let $placeholderAttrs kind="attributes"}
			{if not $_loading}
				hidden
			{/if}

			ref="placeholder"
		{/let}

		<div {$placeholderAttrs}>
			{call .loading /}
		</div>
	</div>
{/template}

/**
 * This renders the component's loading graphic.
 */
{template .loading}
	<div aria-hidden="true" class="loading-icon loading-icon-dotted">
		<span class="loading-icon-indicator"></span>
	</div>
{/template}
