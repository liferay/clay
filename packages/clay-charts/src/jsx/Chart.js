import ChartBase from '../ChartBase';
import {JSXComponent} from 'metal-jsx';

/**
 * Metal Chart component for use with JSX.
 * @extends JSXComponent
 * @extends ChartBase
 */
class ChartJSX extends JSXComponent {
	/**
	 * @inheritDoc
	 */
	render() {
		const id = this.props.id;
		const loading = this.props._loading;

		return (
			<div class="metal-chart-container" {...id && {id}}>
				<div
					class="metal-chart"
					ref="chart"
					{...loading && {hidden: 'hidden'}}
				/>
				<div
					aria-hidden="true"
					class="loading-icon"
					ref="placeholder"
					{...!loading && {hidden: 'hidden'}}
				>
					<span class="loading-animation" />
				</div>
			</div>
		);
	}

	/**
	 * @inheritDoc
	 */
	shouldUpdate() {
		return false;
	}
}

Object.assign(ChartJSX.prototype, ChartBase);

/**
 * @inheritDoc
 */
ChartJSX.PROPS = ChartBase.STATE;

/**
 * @inheritDoc
 */
ChartJSX.prototype._getStateObj = function() {
	return this.props;
};

export {ChartJSX};
export default ChartJSX;
