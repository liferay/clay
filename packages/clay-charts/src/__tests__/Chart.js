import {bb} from 'billboard.js';
import Chart from '../Chart';

beforeAll(() => {
	jest.spyOn(bb, 'generate').mockImplementation();
});

beforeEach(() => {
	bb.generate.mockReset();
});

afterAll(() => {
	bb.generate.mockReset();
	bb.generate.mockRestore();
});

describe('Chart', () => {
	it('should be pass correctly formatted configuration options to billboard.js', done => {
		const chart = new Chart({
			data: [],
		});

		chart.on('chartReady', () => {
			const config = bb.generate.mock.calls[0][0];
			expect(config.bindto).toBe(chart.refs.chart);

			config.bindto = undefined;

			expect(JSON.stringify(config)).toMatchSnapshot();
			done();
		});
	});

	it('should format columns array into billboard.js compatible data', done => {
		const chart = new Chart({
			data: [
				{
					axis: 'y',
					class: 'data1',
					color: 'red',
					data: [1, 2, 3],
					hide: true,
					id: 'data1',
					name: 'Data 1',
					regions: [
						{
							end: 2,
							start: 1,
							style: 'dashed',
						},
					],
					type: 'line',
					x: 'x1',
				},
				{
					axis: 'y2',
					class: 'data2',
					color: 'blue',
					data: [1, 2, 3],
					hide: false,
					id: 'data2',
					name: 'Data 2',
					regions: [
						{
							end: 3,
							start: 2,
							style: 'dashed',
						},
					],
					type: 'spline',
					x: 'x2',
				},
			],
		});

		chart.on('chartReady', () => {
			const config = bb.generate.mock.calls[0][0];
			config.bindto = undefined;

			expect(JSON.stringify(config)).toMatchSnapshot();
			done();
		});
	});

	it('should transform chart new type is passed', done => {
		const transformMock = jest.fn();

		bb.generate.mockReturnValue({
			transform: transformMock,
		});

		const chart = new Chart({
			data: [],
			type: 'line',
		});

		chart.on('chartReady', () => {
			chart.type = 'spline';

			expect(transformMock.mock.calls[0][0]).toBe('spline');
			done();
		});
	});

	it('should rerender chart when new data is passed', done => {
		const loadMock = jest.fn();

		bb.generate.mockReturnValue({
			load: loadMock,
		});

		const chart = new Chart({
			data: [
				{
					id: 'data1',
					data: [1, 2, 3],
				},
			],
		});

		chart.on('chartReady', () => {
			chart.on('dataResolved', () => {
				const config = JSON.stringify(loadMock.mock.calls[0][0]);

				expect(config).toMatchSnapshot();
				done();
			});

			chart.data = [
				{
					id: 'data1',
					data: [1, 2, 3, 4],
				},
			];
		});
	});

	it('should unload removed data when new data is passed', done => {
		const loadMock = jest.fn();

		bb.generate.mockReturnValue({
			load: loadMock,
		});

		const chart = new Chart({
			data: [
				{
					id: 'data1',
					data: [1, 2, 3],
				},
			],
		});

		chart.on('chartReady', () => {
			chart.on('dataResolved', () => {
				const config = JSON.stringify(loadMock.mock.calls[0][0]);

				expect(config).toMatchSnapshot();
				done();
			});

			chart.data = [
				{
					id: 'data2',
					data: [1, 2, 3],
				},
			];
		});
	});
});
