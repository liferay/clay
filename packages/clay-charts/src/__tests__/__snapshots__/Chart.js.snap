// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart should be pass correctly formatted configuration options to billboard.js 1`] = `"{\\"axis\\":{},\\"color\\":{\\"pattern\\":[\\"#4B9BFF\\",\\"#FFD76E\\",\\"#FF5F5F\\",\\"#9CE269\\",\\"#AF78FF\\",\\"#50D2A0\\",\\"#FF73C3\\",\\"#FFB64E\\",\\"#5FC8FF\\"],\\"tiles\\":false},\\"data\\":{\\"colors\\":{},\\"hide\\":false,\\"order\\":\\"desc\\",\\"type\\":\\"line\\",\\"columns\\":[]},\\"grid\\":{\\"x\\":{\\"show\\":true},\\"y\\":{\\"show\\":true}},\\"line\\":{\\"classes\\":[\\"bb-line-dashed-2-2\\",\\"bb-line-dashed-2-3\\",\\"bb-line-dashed-2-4\\",\\"bb-line-dashed-3-2\\",\\"bb-line-dashed-3-3\\",\\"bb-line-dashed-3-4\\",\\"bb-line-dashed-4-2\\",\\"bb-line-dashed-4-3\\",\\"bb-line-dashed-4-4\\"]},\\"point\\":{\\"pattern\\":[\\"circle\\",\\"rectangle\\",\\"<polygon points=\\\\\\"2.5 0 0 5 5 5\\\\\\"></polygon>\\",\\"<polygon points=\\\\\\"2.5 0 0 2.5 2.5 5 5 2.5\\\\\\"></polygon>\\",\\"<polygon points=\\\\\\"0 0 2.5 5 5 0\\\\\\"></polygon>\\",\\"<path d=\\\\\\"M5,5 a1,1 0 1,1 5,0\\\\\\">\\",\\"<path d=\\\\\\"M0,0 a1,1 0 0,0 5,0\\\\\\">\\",\\"<rect width=\\\\\\"5\\\\\\" height=\\\\\\"2.5\\\\\\"></rect>\\",\\"<rect width=\\\\\\"2.5\\\\\\" height=\\\\\\"5\\\\\\"></rect>\\"]},\\"zoom\\":{}}"`;

exports[`Chart should format columns array into billboard.js compatible data 1`] = `"{\\"axis\\":{},\\"color\\":{\\"pattern\\":[\\"#4B9BFF\\",\\"#FFD76E\\",\\"#FF5F5F\\",\\"#9CE269\\",\\"#AF78FF\\",\\"#50D2A0\\",\\"#FF73C3\\",\\"#FFB64E\\",\\"#5FC8FF\\"],\\"tiles\\":false},\\"data\\":{\\"colors\\":{\\"data1\\":\\"red\\",\\"data2\\":\\"blue\\"},\\"hide\\":[\\"data1\\"],\\"order\\":\\"desc\\",\\"type\\":\\"line\\",\\"xs\\":{\\"data1\\":\\"x1\\",\\"data2\\":\\"x2\\"},\\"columns\\":[[\\"data1\\",1,2,3],[\\"data2\\",1,2,3]],\\"axes\\":{\\"data1\\":\\"y\\",\\"data2\\":\\"y2\\"},\\"classes\\":{\\"data1\\":\\"data1\\",\\"data2\\":\\"data2\\"},\\"names\\":{\\"data1\\":\\"Data 1\\",\\"data2\\":\\"Data 2\\"},\\"regions\\":{\\"data1\\":[{\\"end\\":2,\\"start\\":1,\\"style\\":\\"dashed\\"}],\\"data2\\":[{\\"end\\":3,\\"start\\":2,\\"style\\":\\"dashed\\"}]},\\"types\\":{\\"data1\\":\\"line\\",\\"data2\\":\\"spline\\"}},\\"grid\\":{\\"x\\":{\\"show\\":true},\\"y\\":{\\"show\\":true}},\\"line\\":{\\"classes\\":[\\"bb-line-dashed-2-2\\",\\"bb-line-dashed-2-3\\",\\"bb-line-dashed-2-4\\",\\"bb-line-dashed-3-2\\",\\"bb-line-dashed-3-3\\",\\"bb-line-dashed-3-4\\",\\"bb-line-dashed-4-2\\",\\"bb-line-dashed-4-3\\",\\"bb-line-dashed-4-4\\"]},\\"point\\":{\\"pattern\\":[\\"circle\\",\\"rectangle\\",\\"<polygon points=\\\\\\"2.5 0 0 5 5 5\\\\\\"></polygon>\\",\\"<polygon points=\\\\\\"2.5 0 0 2.5 2.5 5 5 2.5\\\\\\"></polygon>\\",\\"<polygon points=\\\\\\"0 0 2.5 5 5 0\\\\\\"></polygon>\\",\\"<path d=\\\\\\"M5,5 a1,1 0 1,1 5,0\\\\\\">\\",\\"<path d=\\\\\\"M0,0 a1,1 0 0,0 5,0\\\\\\">\\",\\"<rect width=\\\\\\"5\\\\\\" height=\\\\\\"2.5\\\\\\"></rect>\\",\\"<rect width=\\\\\\"2.5\\\\\\" height=\\\\\\"5\\\\\\"></rect>\\"]},\\"zoom\\":{}}"`;

exports[`Chart should rerender chart when new columns are passed 1`] = `"{\\"colors\\":{},\\"hide\\":false,\\"order\\":\\"desc\\",\\"type\\":\\"line\\",\\"columns\\":[[\\"data1\\",1,2,3,4]]}"`;

exports[`Chart should unload removed columns when new columns are passed 1`] = `"{\\"colors\\":{},\\"hide\\":false,\\"order\\":\\"desc\\",\\"type\\":\\"line\\",\\"columns\\":[[\\"data2\\",1,2,3]],\\"unload\\":[\\"data1\\"]}"`;
